graphs/views.py

```diff
@@ -193,6 +199,83 @@ def component_coverage(self, component_identifier: str, commit: Commit):
         return filtered_report.totals.coverage
 
 
+class BundleBadgeHandler(APIView, RepoPropertyMixin, GraphBadgeAPIMixin):
+    content_negotiation_class = IgnoreClientContentNegotiation
+
+    permission_classes = [AllowAny]
+
+    extensions = ["svg", "txt"]
+    precisions = ["0", "1", "2"]
+    filename = "bundle-badge"
+
+    def get_object(self, request, *args, **kwargs):
+        # Validate precision query param
+        precision = self.request.query_params.get("precision", "2")
+        precision = int(precision) if precision in self.precisions else 2
+
+        bundle_size_bytes = self.get_bundle_size()
+
+        if self.kwargs.get("ext") == "txt":
+            return (
+                "unknown"
+                if bundle_size_bytes is None
+                else format_bundle_bytes(bundle_size_bytes, precision)
+            )
+
+        return get_bundle_badge(bundle_size_bytes, precision)
+
+    def get_bundle_size(self) -> int | None:
+        try:
+            repo = self.repo
+        except Http404:
+            log.warning("Repo not found", extra=dict(repo=self.kwargs.get("repo_name")))
+            return None
+
+        if repo.private and repo.image_token != self.request.query_params.get("token"):
+            log.warning(
+                "Token provided does not match repo's image token",
+                extra=dict(repo=repo),
+            )
+            return None
+
+        branch_name = self.kwargs.get("branch") or repo.branch
+        branch = Branch.objects.filter(
+            name=branch_name, repository_id=repo.repoid
+        ).first()
+
+        if branch is None:
+            log.warning(
+                "Branch not found", extra=dict(branch_name=branch_name, repo=repo)
+            )
+            return None
```

Comment:
Should we throw a 404 here, when a commit is not found, or if the bundle is not found?

Reply:
Coverage follows this pattern, we turn any None into a badge that says `unknown`. I think the motivation for this is so that the badge will always show *something* instead of looking broken

Reply:
Ahhh i see i see, sounds good 👍